datadog_checks_base/datadog_checks/base/checks/win/wmi/__init__.py

```diff
@@ -43,11 +51,33 @@ class WinWMICheck(AgentCheck):
     """
 
     def __init__(self, *args, **kwargs):  # To support optional agentConfig
+        # type: (List[Any], Dict[str, Any]) -> None
         super(WinWMICheck, self).__init__(*args, **kwargs)
-        self.wmi_samplers = {}
-        self.wmi_props = {}
+
+        if not self.instance:
+            # This should not happen and is here for legacy purposes.
+            # Once all checks are updated to use the agent 6 signature we can change self.instance type in base class
+            # to not be optional and remove this check.
+            raise ConfigurationError("No instance configuration provided")
```

Comment:
If the motivation for this check is preventing mypy complains, should we drop this once https://github.com/DataDog/integrations-core/pull/6350 has been merged?